pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1689,7 +1689,14 @@ def _check_unnecessary_dict_index_lookup(
                         and subscript == subscript.parent.target
                     ):
                         # Ignore this subscript if it is the target of an assignment
-                        continue
+                        if subscript.as_string() == subscript.parent.value.as_string():
+                            # Fire error as d[k] += d[k] has an unnecessary index lookup
+                            self.add_message(
+                                "unnecessary-dict-index-lookup",
+                                node=subscript,
+                                args=(node.target.elts[1].as_string()),
+                            )
+                        return  # Early termination; after reassignment dict index lookup will be necessary
```

Comment:
Move the comment on an empty line above?

Reply:
yeah sure (:

Reply:
```suggestion
                        # Early termination; after reassignment dict index lookup will be necessary
                        return 
```

Reply:
fixed in 14b6db0